File: Challenges/450_DSA_Cracker/017_Best_Time_To_Buy_Sell.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy = Integer.MAX_VALUE;
        int l = prices.length;
        int maxProfit = 0, profit;

        for(int i=0; i<l; i++) {
            buy = Math.min(buy, prices[i]);
            profit = prices[i] - buy;
            maxProfit = Math.max(profit, maxProfit);
        }

        return maxProfit;
    }
}

File: Challenges/450_DSA_Cracker/Array/017_Best_Time_To_Buy_Sell.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minBuy = 10000
        maxProfit = 0
        for i in prices : 
            minBuy = min(i, minBuy)
            maxProfit = max(maxProfit, i-minBuy)
        return maxProfit
